packages/headless/src/features/commerce/product-listing/product-listing-slice.ts

```diff
@@ -57,6 +77,22 @@ export const productListingReducer = createReducer(
             )
           )
         );
+        state.results = state.results.concat(
+          action.payload.response.results.map(
+            (result, index): Result =>
+              result.resultType === ResultType.SPOTLIGHT
+                ? preprocessSpotlightContent(
+                    result,
+                    paginationOffset + index + 1,
+                    action.payload?.response.responseId
+                  )
+                : preprocessProduct(
+                    result,
+                    paginationOffset + index + 1,
+                    action.payload?.response.responseId
+                  )
+          )
```

Comment:
This is hard to read IMO.
I'd merge the preprocess function (or create a "preprocessResult" one) that do the branching based on the resultype.